LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/ModuleConnection/ModuleConnectionHost.cs

```diff
@@ -0,0 +1,207 @@
+           ILoRaDeviceRegistry loRaDeviceRegistry,
+           ILoraModuleClient loraModuleClient)
+        {
+            this.networkServerConfiguration = networkServerConfiguration ?? throw new ArgumentNullException(nameof(networkServerConfiguration));
+            this.classCMessageSender = defaultClassCDevicesMessageSender ?? throw new ArgumentNullException(nameof(defaultClassCDevicesMessageSender));
+            this.loRaDeviceRegistry = loRaDeviceRegistry ?? throw new ArgumentNullException(nameof(loRaDeviceRegistry));
+            this.loRaModuleClient = loraModuleClient;
+        }
+
+        public async Task CreateAsync()
+        {
+            ITransportSettings[] settings = { new AmqpTransportSettings(Microsoft.Azure.Devices.Client.TransportType.Amqp_Tcp_Only) };
+
+            await this.loRaModuleClient.CreateFromEnvironmentAsync(settings);
+            await InitModuleAsync();
+
+        }
+
+        internal async Task InitModuleAsync()
+        { 
+            // Obsolete, this should be removed as part of #456
+            Logger.Init(new LoggerConfiguration
+            {
+                ModuleClient = this.loRaModuleClient.GetModuleClient(),
+                LogLevel = LoggerConfiguration.InitLogLevel(networkServerConfiguration.LogLevel),
+                LogToConsole = networkServerConfiguration.LogToConsole,
+                LogToHub = networkServerConfiguration.LogToHub,
+                LogToUdp = networkServerConfiguration.LogToUdp,
+                LogToUdpPort = networkServerConfiguration.LogToUdpPort,
+                LogToUdpAddress = networkServerConfiguration.LogToUdpAddress,
+                GatewayId = networkServerConfiguration.GatewayID
+            });
+
+            if (networkServerConfiguration.IoTEdgeTimeout > 0)
+            {
+                this.loRaModuleClient.OperationTimeoutInMilliseconds = networkServerConfiguration.IoTEdgeTimeout;
+                Logger.Log($"Changing timeout to {networkServerConfiguration.IoTEdgeTimeout} ms", LogLevel.Debug);
+            }
+
+            Logger.Log("Getting properties from module twin...", LogLevel.Information);
+            Twin moduleTwin;
+            try
+            {
+                moduleTwin = await this.loRaModuleClient.GetTwinAsync();
+            }
+            catch (IotHubCommunicationException)
+            {
+                Logger.Log($"There was a critical problem with the IoT Hub in getting the module twins.", LogLevel.Error);
+                throw;
+            }
+
+            var moduleTwinCollection = moduleTwin.Properties.Desired;
+
+            if (!TryUpdateConfigurationWithDesiredProperties(moduleTwinCollection))
+            {
+                Logger.Log($"The initial configuration of the facade function could not be found in the desired properties", LogLevel.Error);
+                throw new ArgumentException("Could not get Facade information from module twin");
```

Comment:
`ArgumentException` is reserved for method arguments whereas here it's being used for a configuration error at run-time. Also, aren't we over-logging here? I think we are already logging in `TryUpdateConfigurationWithDesiredProperties`?

Reply:
The problem is that from inside the method I don't know if that is the getwin from startup or the one from the Ondesired property change trigger. Therefore I need to defer some logs from outside the function. If you think it's too much I could remove some of the logging in the method, as the levels are generally debug, I don't see much of an issue here,

Reply:
Changed to ConfigurationErrorsException